scripts/firebase-config-setup.sh

```diff
@@ -28,23 +28,20 @@ usage() {
   exit 1
 }
 
-readonly PROJECT_ID="$1"
+export readonly PROJECT_ID="$1"
 readonly APP_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
-readonly FIREBASECONFIG_JS="${APP_ROOT}/webui/firebaseConfig.js"
+readonly FIREBASECONFIG_SAMPLE="${APP_ROOT}/webui/firebaseConfig.sample.ts"
+readonly FIREBASECONFIG_JS="${APP_ROOT}/webui/firebaseConfig.ts"
 
 if [[ "$#" -ne 2 ]]; then
   usage
 fi
 
-readonly API_KEY="$2"
+export readonly API_KEY="$2"
+echo ${API_KEY}
 
-cat > "${FIREBASECONFIG_JS}" << FIREBASECONFIG
-export const firebaseConfig = {
-  "projectId": "${PROJECT_ID}",
-  "apiKey": "${API_KEY}",
-  "authDomain": "${PROJECT_ID}.firebaseapp.com"
-};
-FIREBASECONFIG
+cat ${FIREBASECONFIG_SAMPLE}
+cat ${FIREBASECONFIG_SAMPLE} | envsubst > ${FIREBASECONFIG_JS}
```

Comment:
`envsubst` may not necessary be present in all runtime environments -- it is installed as part of `gettext`. Probably better not to depend on it directly.

Reply:
What should we do then? `sed`? I don't want to get back to duplicating the template here.

Reply:
Could revert to how it was was done before. It's only a few lines. IMO it's easier to read than `sed` 🤷‍♂️

Reply:
how it was done before was copying the content of `firebaseConfig.js.sample` here. I really don't think duplicating that code in two places is a better idea. I already almost forgot to update this part this time.

Reply:
If we are going to keep `envsubst`, it needs to be called out in the prerequisites section of the README as a required dependency. But I would strongly prefer that we didn't add an additional dependency just for this script -- `sed` may be uglier, but it's not as bad as asking every user to install `gettext` IMHO.

Reply:
changed it to `sed` instead.